diff --git a/include/mbedtls/mbedtls_config.h b/include/mbedtls/mbedtls_config.h
index d283136..4e31b2b 100644
--- a/include/mbedtls/mbedtls_config.h
+++ b/include/mbedtls/mbedtls_config.h
@@ -540,7 +540,7 @@
  *
  * Uncomment to use your own hardware entropy collector.
  */
-//#define MBEDTLS_ENTROPY_HARDWARE_ALT
+#define MBEDTLS_ENTROPY_HARDWARE_ALT
 
 /**
  * \def MBEDTLS_AES_ROM_TABLES
@@ -1206,7 +1206,7 @@
  *
  * Uncomment this macro to disable the built-in platform entropy functions.
  */
-//#define MBEDTLS_NO_PLATFORM_ENTROPY
+#define MBEDTLS_NO_PLATFORM_ENTROPY
 
 /**
  * \def MBEDTLS_ENTROPY_FORCE_SHA256
@@ -1521,7 +1521,7 @@
  *
  * Enable the checkup functions (*_self_test).
  */
-#define MBEDTLS_SELF_TEST
+//#define MBEDTLS_SELF_TEST
 
 /**
  * \def MBEDTLS_SHA256_SMALLER
@@ -3773,7 +3773,7 @@
  *
  * Module:  library/timing.c
  */
-#define MBEDTLS_TIMING_C
+//#define MBEDTLS_TIMING_C
 
 /**
  * \def MBEDTLS_VERSION_C
diff --git a/library/entropy.c b/library/entropy.c
index 7dcf067..76c4429 100644
--- a/library/entropy.c
+++ b/library/entropy.c
@@ -24,6 +24,20 @@
 
 #define ENTROPY_MAX_LOOP    256     /**< Maximum amount to loop before error */
 
+int mbedtls_hardware_poll(void *data,
+                          unsigned char *output, size_t len, size_t *olen)
+{
+    (void)data;
+
+    uint32_t Wifi_Random(void); // Internal function of DSWiFi
+
+    for (size_t i = 0; i < len; i++)
+        output[i] = Wifi_Random();
+    *olen = len;
+
+    return 0;
+}
+
 void mbedtls_entropy_init(mbedtls_entropy_context *ctx)
 {
     ctx->source_count = 0;
diff --git a/library/platform_util.c b/library/platform_util.c
index 19ef07a..a607c03 100644
--- a/library/platform_util.c
+++ b/library/platform_util.c
@@ -258,6 +258,10 @@ mbedtls_ms_time_t mbedtls_ms_time(void)
     return current_ms;
 }
 #else
-#error "No mbedtls_ms_time available"
+#include <time.h>
+mbedtls_ms_time_t mbedtls_ms_time(void)
+{
+    return time(NULL) * 1000;
+}
 #endif
 #endif /* MBEDTLS_HAVE_TIME && !MBEDTLS_PLATFORM_MS_TIME_ALT */
